impls/src/adapters/emoji.rs

```diff
@@ -13,1041 +13,174 @@
 // limitations under the License.
 
 use bitvec::prelude::*;
-use emoji::{self, Emoji};
 
 use crate::libwallet::{Error, ErrorKind, Slate, SlateVersion, VersionedSlate};
 
+extern crate flate2;
+use super::emoji_map::{
+	DEFLATE_METHOD, EMOJI_DIVIDER, EMOJI_MAP, GZIP_METHOD, VERSION_0, VERSION_1, ZLIB_METHOD,
+};
+
+use super::compress::{compress, decompress, CompressionFormat};
+use std::collections::HashMap;
+
+/// Globally controls which type of transaction we will consider, compressed or normal
+static TYPE_TRANSACTION: &str = "compress";
+
+/// The default method to compress and decompress in this version of Epic
+const COMPRESS_METHOD: CompressionFormat = CompressionFormat::Gzip;
+
+/// The default version of the emoji in this version of Epic
+const EMOJI_VERSION: u16 = 1;
+
+/// Saves all information and the type of compressor and which version of the epic for the emoji we are using
+#[derive(Debug)]
+struct Header {
+	/// Method to compress/decompress
+	algo: CompressionFormat,
+	/// Version of emoji in Epic
+	version: u16,
+}
+
+/// Get a Vec of String -> Header
+impl From<Vec<String>> for Header {
+	fn from(input: Vec<String>) -> Self {
+		// First information, algo
+		let algo_str = &input[0];
+		// Second position, version
+		let version_str = &input[1];
+
+		// Get the Enum from str
+		let algo = match algo_str.as_ref() {
+			"gzip" => CompressionFormat::Gzip,
+			"zlib" => CompressionFormat::Zlib,
+			"deflate" => CompressionFormat::Deflate,
+			_ => panic!("Invalid compression format!"),
+		};
+
+		// Get the version from str
+		let version = match version_str.parse() {
+			Ok(version) => version,
+			Err(_) => panic!("Invalid version number!"),
+		};
+
+		// return the Header
+		Header { algo, version }
+	}
+}
```

Comment:
You could fully convert it to a string in this function

Reply:
It's more efficient to pass straight to String but it's a safer way to make this transition from Emoji->Struct ->String (in another step).

I'll make the change to be more efficient.